Add tutorial template (#229)
thegooddocsproject/templates: tutorial/tutorial-template.md

# Title

{Before using this template, read the accompanying [tutorial template guide](tutorial-template-guide)}.

## Overview

> In this tutorial, you'll learn how to {top learning objective} using {product}. This tutorial is intended for {audience}. It assumes you have basic knowledge of:

Reviewer: Is there another word I can use instead of "top learning objective?" Main learning objective?

Like the "main" or "top" learning objective could be _create components in React_. And then the learning objectives you list below might include:

* Modify arguments that are passed to React components
* Describe the differences between functional and class components
* ....

Open to any suggestions here.

Reviewer 2: For me, I'd maybe say: "In this tutorial you'll learn how to {insert brief description of the main tutorial task}." I'd leave it at that because your accompanying guide goes into a lot more detail about it and they can just refer to that.

Actually, I think it would be good to check that this section closely matches your guide section. I haven't done a direct comparison, but I feel like there might be subtle differences when I think of it from memory?

Reviewer: @barbaricyawps "Brief description of the main tutorial task" might be exactly what I'm looking for. That first line can briefly describe what you're going to do, and later you can list the specific learning objectives users will achieve by the end of the tutorial. 

I can update my tutorial template...it is a tad out of sync. Thank you!

* Concept 1
* Concept 2
* Concept 3...

By the end of this tutorial, you'll be able to:

* Learning objective 1
* Learning objective 2
* Learning objective 3...

## Background

{This section is optional. Feel free to use some of the text below to help you get started.}

* {product} is a {product type} that you can use to {common use case}... 

> * {product} provides many of the same features as {competitors}, but with {feature}, you can...

Reviewer 3: Suggestion:
```suggestion
* {product} provides many of the same features as {competitors}, but with {feature}. You can...
```

* Using {feature} enables you to {pain point}...

## Before you begin 

> {Use this section to tell users about any steps they need to complete before they start the tutorial.}

Reviewer 3: Suggestion:
I'm thinking pre-requisites might be more than following steps.
```suggestion
{Use this section to tell users about any prerequisites needed before they start the tutorial, such as:
* Expected prior knowledge.
* Software or hardware to obtain.
* Environments to set up and configure.
* Access codes to obtain.
}
```

> Before you start the tutorial, complete these steps:

Reviewer 3: Suggestion: As above, prerequisites can be more than following steps. (My suggested change could likely be improved upon.)
```suggestion
Before you start the tutorial, you should:
```

* Prerequisite 1
* Prerequisite 2
* Prerequisite 3...

## {Task name}

To get started, {the first thing your user should do}.

1. {Write the step here. Use a verb to start.}

    {Explanatory text}

    {Optional: Code sample or screenshot that helps your users complete this step}

    {Optional: Result}

2. {Write the step here. Use a verb to start.}
   
   a. {Substep 1}

   b. {Substep 1} 

## Summary

{Use this section to summarize what the user learned in the tutorial.}

In this tutorial, you learned how to:

* Summary point 1
* Summary point 2
* Summary point 3..

## Next steps

{Use this section to share links to related tutorials, videos, or other documentation}.

Consider completing some other common tasks using {feature}:

* Task 1
* Task 2

> * Task 3...

Reviewer 2: **Praise:** I really love how clean and bare-bones your latest version of the tutorial is. It makes it a lot more useable. Great work!

Reviewer 3: +1 Great work

# Title

{Before using this template, read the accompanying [tutorial template guide](tutorial-template-guide)}.

## Overview

In this tutorial, you'll learn how to {insert brief description of the main tutorial task}. This tutorial is intended for {audience}. It assumes you have basic knowledge of:

* Concept 1
* Concept 2
* Concept 3...

By the end of this tutorial, you'll be able to:

* Learning objective 1
* Learning objective 2
* Learning objective 3...

## Background

{This section is optional. Feel free to use some of the text below to help you get started.}

* {product} is a {product type} that you can use to {common use case}... 

> * {product} provides many of the same features as {competitors}, but with {feature}. You can...

Reviewer 2: ```suggestion
* {product} provides many of the same features as {competitors}, but with {feature}, you can...
```
